pkg/gofr/subscriber.go

```diff
@@ -24,39 +22,56 @@ func newSubscriptionManager(c *container.Container) SubscriptionManager {
 	}
 }
 
-func (s *SubscriptionManager) startSubscriber(topic string, handler SubscribeFunc) {
+func (s *SubscriptionManager) startSubscriber(ctx context.Context, topic string, handler SubscribeFunc) error {
 	// continuously subscribe in an infinite loop
 	for {
-		msg, err := s.container.GetSubscriber().Subscribe(context.Background(), topic)
-		if msg == nil {
-			continue
+		select {
+		case <-ctx.Done():
+			s.container.Logger.Infof("shutting down subscriber for topic %s", topic)
+			return nil
+		default:
+			err := s.handleSubscription(ctx, topic, handler)
+			if err != nil {
+				return err
+			}
 		}
+	}
+}
 
-		if errors.Is(err, kafka.ErrConsumerGroupNotProvided) {
-			s.container.Logger.Errorf("cannot subscribe as consumer_id is not provided in configs")
-			return
-		} else if err != nil {
-			s.container.Logger.Errorf("error while reading from topic %v, err: %v", topic, err.Error())
-			continue
-		}
+func (s *SubscriptionManager) handleSubscription(parentCtx context.Context, topic string, handler SubscribeFunc) error {
+	msg, err := s.container.GetSubscriber().Subscribe(parentCtx, topic)
```

Comment:
```suggestion
func (s *SubscriptionManager) handleSubscription(ctx context.Context, topic string, handler SubscribeFunc) error {
	msg, err := s.container.GetSubscriber().Subscribe(ctx, topic)
```

Let's go back to classic naming for a context

Reply:
Sure